The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26579441864/job/78308031501) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/78308031501) [(plain)](https://github.com/rust-lang/rust/commit/7023463cafcea4dbe285c6dcfa50f535d5116a13/checks/78308031501/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
1 error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
-   --> $DIR/unpin-for-future.rs:18:1
+   --> $DIR/unpin-for-future.rs:19:1
3    |
4 LL | impl Unpin for MyFut {}
5    | ^^^^^^^^^^^^^^^-----

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/impl-trait/unpin-for-future.rs:19:1
+   --> $DIR/unpin-for-future.rs:19:1
---
To only update this specific test, also pass `--test-args impl-trait/unpin-for-future.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/impl-trait/unpin-for-future.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/impl-trait/unpin-for-future" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2024"
stdout: none
--- stderr -------------------------------
error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
##[error]  --> /checkout/tests/ui/impl-trait/unpin-for-future.rs:19:1
   |
LL | impl Unpin for MyFut {}
   | ^^^^^^^^^^^^^^^-----
   |                |
   |                type alias impl trait is treated as if it were foreign, because its hidden type could be from a foreign crate
   |
   = note: impl doesn't have any local type before any uncovered type parameters
```

</details>

--- a/tests/ui/impl-trait/unpin-for-future.rs
+++ b/tests/ui/impl-trait/unpin-for-future.rs
@@ -1,0 +1,22 @@
+//@ edition:2024
+//
+// Tests that you can't implement Unpin for a compiler-generated future using TAIT.
+
+#![feature(type_alias_impl_trait)]
+
+use core::marker::PhantomPinned;
+use core::pin::Pin;
+
+type MyFut = impl Future<Output = ()>;
+
+async fn my_async_fn() {}
+
+#[define_opaque(MyFut)]
+fn fut() -> MyFut {
+    my_async_fn()
+}
+
+impl Unpin for MyFut {}
+//~^ ERROR: only traits defined in the current crate can be implemented for arbitrary types
+
+fn main() {}

--- a/tests/ui/impl-trait/unpin-for-future.stderr
+++ b/tests/ui/impl-trait/unpin-for-future.stderr
@@ -1,0 +1,15 @@
+error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
+  --> $DIR/unpin-for-future.rs:19:1
+   |
+LL | impl Unpin for MyFut {}
+   | ^^^^^^^^^^^^^^^-----
+   |                |
+   |                type alias impl trait is treated as if it were foreign, because its hidden type could be from a foreign crate
+   |
+   = note: impl doesn't have any local type before any uncovered type parameters
+   = note: for more information see https://doc.rust-lang.org/reference/items/implementations.html#orphan-rules
+   = note: define and implement a trait or new type instead
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0117`.

